webpack/assets/javascripts/react_app/routes/RoutingService.js

```diff
@@ -0,0 +1,42 @@
+import { visit } from '../../foreman_navigation';
+
+let currentPath = window.location.href;
+
+/**
+ * a Helper function for rendering a route
+ * @param {Function} renderFn - a component's rendering function
+ * @param {Object} props - routing props
+ * @param {Function} beforeRenderingCallback - a callback to run before the rendering, if it returns false the rendering will terminate
+ */
+export const renderRoute = (
+  renderFn,
+  props,
+  beforeRenderingCallback = () => true
```

Comment:
can you use `noop` here instead?

Reply:
I tried but it doesn't work in this case